plugin/src/main/java/org/opensearch/sql/plugin/catalog/PPLCatalogService.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.sql.plugin.catalog;
+
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import java.io.IOException;
+import java.io.InputStream;
+import java.security.PrivilegedExceptionAction;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import org.apache.commons.math3.util.Pair;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.sql.catalog.CatalogService;
+import org.opensearch.sql.executor.ExecutionEngine;
+import org.opensearch.sql.opensearch.security.SecurityAccess;
+import org.opensearch.sql.opensearch.setting.OpenSearchSettings;
+import org.opensearch.sql.storage.StorageEngine;
+
+
+public class PPLCatalogService implements CatalogService {
```

Comment:
Why catalog bind to PPL?

Reply:
I have initially implemented the same but got into some problems with dependency on PPL components. Will take a look again.

Reply:
Changed the class name CatalogServiceImpl.